### PR TITLE
 Convert mockwebserver/QueueDispatcher and Dispatcher to Kotlin

### PR DESCRIPTION
**Pull request by TangHuaiZhe**

### CODE REVIEW

File: mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+package okhttp3.mockwebserver
+
+import java.net.HttpURLConnection
+import java.net.HttpURLConnection.HTTP_UNAVAILABLE
+import java.util.concurrent.BlockingQueue
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.logging.Logger
+
+/**
+ * Default dispatcher that processes a script of responses. Populate the script by calling [ ][.enqueueResponse].
```
Comment (JakeWharton) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt]: Fix reference

File: mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  open fun enqueueResponse(response: MockResponse) {
+    responseQueue.add(response)
+  }
+
+  override fun shutdown() {
+    responseQueue.add(DEAD_LETTER)
+  }
+
+  open fun setFailFast(failFast: Boolean) {
+    val failFastResponse = if (failFast)
```
Comment (JakeWharton) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt]: curlies

File: mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  override fun peek(): MockResponse {
+    val peek = responseQueue.peek()
+    if (peek != null) return peek
+    if (failFastResponse != null) {
+      return failFastResponse!!
+    } else {
+      return super.peek()
+    }
```
Comment (JakeWharton) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt]:
```kotlin
return responseQueue.peek() ?: failFastResponse ?: super.peek()
```

Comment (TangHuaiZhe) [reply mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt]: > ```kotlin
> return responseQueue.peek() ?: failFastResponse ?: super.peek()
> ```

Oh! Intelligent!

File: mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt (hunk 1 above)
Comment (TangHuaiZhe) [reply mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt]: Fixed

File: mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt (hunk 2 above)
Comment (TangHuaiZhe) [reply mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt]: Fixed

**Pull request merged**